openmetadata-ui/src/main/resources/ui/src/components/Modals/EntityNameModal/EntityNameModal.component.tsx

```diff
@@ -11,22 +11,16 @@
  *  limitations under the License.
  */
 import { Button, Form, Input, Modal, Typography } from 'antd';
-import { EntityReference } from 'generated/type/entityReference';
 import React, { useEffect } from 'react';
 import { useTranslation } from 'react-i18next';
+import { EntityNameModalProps } from './EntityNameModal.interface';
 
-interface Props {
-  visible: boolean;
-  onCancel: () => void;
-  onSave: (obj: { name: string; displayName: string }) => void;
-  entity: EntityReference;
-}
-
-const EntityNameModal: React.FC<Props> = ({
+const EntityNameModal: React.FC<EntityNameModalProps> = ({
   visible,
   entity,
   onCancel,
   onSave,
+  disableNameUpdate = false,
```

Comment:
```suggestion
  allowRename = false,
```

changing name is considered as renaming
Let's have prop name like that, If renaming is not supported name field will be disbled
Add comments in the code for a clear picture